File: server/core/api/position.go
package api

import "letstalk/server/data"

type Role struct {
	Id   data.TRoleID `json:"id" binding:"required"`
	Name string       `json:"name" binding:"required"`
}

type Organization struct {
	Id   data.TOrganizationID  `json:"id" binding:"required"`
	Name string                `json:"name" binding:"required"`
	Type data.OrganizationType `json:"type" binding:"required"`
}

type AddUserPositionRequest struct {
	RoleId           *data.TRoleID         `json:"roleId"`
	RoleName         *string               `json:"roleName"`
	OrganizationId   *data.TOrganizationID `json:"organizationId"`
	OrganizationName *string               `json:"organizationName"`
	StartDate        string                `json:"startDate" binding:"required"`
	EndDate          *string               `json:"endDate"`
}

type RemoveUserPositionRequest struct {
	UserPositionId data.TUserPositionID `json:"userPositionId" binding:"required"`
}

File: letstalk/src/models/position.ts
export interface Role {
  readonly id: number;
  readonly name: string;
};

type OrganizationType = string;

const	ORGANIZATION_TYPE_COMPANY:      OrganizationType = "COMPANY";
const	ORGANIZATION_TYPE_CLUB:         OrganizationType = "CLUB";
const	ORGANIZATION_TYPE_SPORTS_TEAM:  OrganizationType = "SPORTS_TEAM";
const	ORGANIZATION_TYPE_UNDETERMINED: OrganizationType = "UNDETERMINED";

export interface Organization {
  readonly id: number;
  readonly name: string;
  readonly type: OrganizationType;
};

export {
  ORGANIZATION_TYPE_COMPANY,
  ORGANIZATION_TYPE_CLUB,
  ORGANIZATION_TYPE_SPORTS_TEAM,
  ORGANIZATION_TYPE_UNDETERMINED,
};

export interface Position {
  readonly roleId: number;
  readonly roleName: string;
  readonly organizationId: number;
  readonly organizationName: string;
  readonly organizationType: OrganizationType,
}

export interface UserPosition extends Position {
  readonly id: number;
  readonly startDate: string;
  readonly endDate: string | null;
};
